Keas.Mvc/Controllers/Api/AccessController.cs

```diff
@@ -213,5 +237,43 @@ public async Task<IActionResult> Delete(int id)
             }
 
         }
+
+        /// <summary>
+        /// Return history records
+        /// Defaults to a max of 5 records returned
+        /// </summary>
+        /// <param name="id"></param>
+        /// <param name="max">the max number of record to take. Defaults to 5</param>
+        /// <returns></returns>
+        [HttpGet("{id}")]
+        [ProducesResponseType(typeof(IEnumerable<History>), StatusCodes.Status200OK)]
+        public async Task<IActionResult> GetHistory(int id, int max = 5)
+        {
+            var history = await _context.Histories
+                .Where(x => x.AssetType == "Access" && x.Access.Team.Slug == Team && x.AccessId == id)
+                .OrderByDescending(x => x.ActedDate)
+                .Take(max)
+                .AsNoTracking().ToListAsync();
+
+            return Json(history);
+        }
+
+        /// <summary>
+        /// Return all history records
+        /// </summary>
+        /// <param name="id"></param>
+        /// <returns></returns>
+        [HttpGet("{id}")]
+        [ProducesResponseType(typeof(IEnumerable<History>), StatusCodes.Status200OK)]
+        public async Task<IActionResult> GetFullHistory(int id)
```

Comment:
do we inactivate history?  And if so, do we really want to return inactive history?  I don't see why this is needed

Reply:
@srkirkland Not the history, but the access that has the history.